tests/functions/test_apply_over.py

```diff
@@ -0,0 +1,13 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+
+@pytest.mark.functions
+def test_apply_over():
+    df = pd.DataFrame({"a": ["dog", "dog", "cat", "cat"], "b": [1, 2, 3, 4]})
+
+    res = df.apply_over(func=np.max, col="b", by="a", name="max")
+    exp = pd.concat((df, pd.Series([2, 2, 4, 4], name="max")), axis=1)
+
+    assert res == exp
```

Comment:
This test should use `pd.testing.assert_frame_equal`, which is the proper way to conduct the test. Could you change this for the test for `take_first`?

Reply:
Actually, don't worry about `take_first`; I'll handle that one.